tests/auth_tests/test_auth_backends.py

```diff
@@ -472,6 +476,10 @@ def test_has_perms(self):
         self.assertIs(self.user1.has_perms(['anon'], TestObj()), True)
         self.assertIs(self.user1.has_perms(['anon', 'perm'], TestObj()), False)
 
+    def test_has_perms_cannot_be_str(self):
+        with self.assertRaises(TypeError):
```

Comment:
Please assert an error message, e.g.
```suggestion
        msg = 'perm_list must be a list or tuple.'
        with self.assertRaisesMessage(ValueError, msg):
```